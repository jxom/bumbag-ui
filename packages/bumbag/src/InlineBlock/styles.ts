import { cssClass } from '../styled';
import { theme } from '../utils';

export const InlineBlock = (styleProps) => cssClass`
  display: inline-block;

  & {
    ${theme(styleProps.themeKey, `styles.base`)(styleProps)};
  }
`;
