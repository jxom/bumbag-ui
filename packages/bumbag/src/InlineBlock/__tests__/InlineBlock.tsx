import * as React from 'react';
import { renderHook } from '@testing-library/react-hooks';
import { InlineBlock } from '../InlineBlock';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    const { container } = render(<InlineBlock>Hello world</InlineBlock>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with CSS props', () => {
    const { container } = render(
      <InlineBlock position="absolute" top="0">
        Hello world
      </InlineBlock>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('composition', () => {
  describe('as', () => {
    it('should render correctly', () => {
      const { container } = render(<InlineBlock use="p">Hello world</InlineBlock>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('hook', () => {
    it('should return with InlineBlock props', () => {
      const { result } = renderHook(() => InlineBlock.useProps());
      expect(result.current).toMatchSnapshot();
    });
  });

  describe('render props', () => {
    it('should render correctly', () => {
      const { container } = render(
        <InlineBlock>{(InlineBlockProps) => <div {...InlineBlockProps}>Hello world</div>}</InlineBlock>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});

describe('theming', () => {
  it('InlineBlock.base should render correctly', () => {
    const { container } = render(<InlineBlock>hello world</InlineBlock>, {
      theme: { InlineBlock: { styles: { base: { backgroundColor: 'red' } } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('defaultProps', () => {
  it('should render correctly for className', () => {
    const { container } = render(<InlineBlock>hello world</InlineBlock>, {
      theme: { InlineBlock: { defaultProps: { className: 'test' } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
