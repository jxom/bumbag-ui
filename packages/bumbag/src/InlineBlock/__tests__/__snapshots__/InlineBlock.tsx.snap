// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`composition as should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  display: inline-block;
}

<p
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-InlineBlock"
>
  Hello world
</p>
`;

exports[`composition hook should return with InlineBlock props 1`] = `
Object {
  "className": "css-1cwcv4r css-4lfl7m css-1yuzjtp fp-Box fp-InlineBlock",
}
`;

exports[`composition render props should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  display: inline-block;
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-InlineBlock"
>
  Hello world
</div>
`;

exports[`defaultProps should render correctly for className 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  display: inline-block;
}

<div
  class="emotion-0 emotion-1 emotion-2 test fp-Box fp-InlineBlock"
>
  hello world
</div>
`;

exports[`props should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  display: inline-block;
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-InlineBlock"
>
  Hello world
</div>
`;

exports[`props should render correctly with CSS props 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  display: inline-block;
}

.emotion-2 {
  position: absolute !important;
  top: 0 !important;
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-InlineBlock"
>
  Hello world
</div>
`;

exports[`theming InlineBlock.base should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  display: inline-block;
}

.emotion-0.emotion-0 {
  background-color: #ff0000;
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-InlineBlock"
>
  hello world
</div>
`;
